test/box-luatest/gh_9406_fix_use_after_free_on_shutdown_test.lua

```diff
@@ -0,0 +1,24 @@
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_shutdown_with_lingering_fiber_join = function()
+    local script = [[
+        local fiber = require('fiber')
+
+        local f = nil
+        fiber.create(function()
+            fiber.sleep(0.1)
+            fiber.join(f)
+        end)
+        f = fiber.create(function()
```

Comment:
You should use `fiber.new` here because `fiber.create` yields.

Reply:
Done.

```diff
diff --git a/test/app-luatest/fiber_test.lua b/test/app-luatest/fiber_test.lua
index bc5237d35..8a681b48e 100644
--- a/test/app-luatest/fiber_test.lua
+++ b/test/app-luatest/fiber_test.lua
@@ -41,10 +41,12 @@ g.test_gh_9406_shutdown_with_lingering_fiber_join = function()
             fiber.join(f)
         end)
-        f = fiber.create(function()
+        f = fiber.new(function()
             fiber.sleep(1000)
         end)
         f:set_joinable(true)
```